prereise/gather/data/remap_ba_area/remap_ba_area.py

```diff
@@ -0,0 +1,270 @@
+import json
+import os
+import shutil
+
+import geopandas as gpd
+import matplotlib.pyplot as plt
+import pandas as pd
+import requests
+
+from prereise.gather.data.remap_ba_area.map_data import (
+    miso_name_map,
+    spp_name_map,
+    states_to_map_with_counties,
+)
+
+
+def get_full_filename(file_name):
+    return os.path.join(os.path.dirname(__file__), file_name)
+
+
+def compose_ba_map(output_shape_file="remap_ba_area.shp"):
+    """
+    Create the control area and subregion map. For CAs heavily overlapped, we define several zones and match the CAs to
+    these zones. The ca_source name is either mapped to one zone in BA_map_name, or allocated to multiple zones in
+    BA_map_name with fractions based on the area.
+    The function generates the remap_ba_area shape file and remap_ba_area.tif file.
+
+    :param str output_shape_file: output shape file name
+    :return: (*geopandas.GeoDataFrame*)  -- the final control zone map
+    """
+
+    with open(get_full_filename("../BA_County_map.json")) as json_file:
+        ba2county_dict = json.load(json_file)
+    # Add two missing counties
+    # this may be fixed in future updates of BA_County_map.json
+
+    # Turn dict into pandas table
+    county2ba_dict = {
+        county: ba
+        for (ba, ba_counties) in ba2county_dict.items()
+        for county in ba_counties
+    }
+    county2ba = pd.Series(county2ba_dict).to_frame().reset_index()
+    county2ba.columns = ["index", "BA"]
+    county2ba[["county", "state"]] = county2ba["index"].str.split(
+        "__", n=1, expand=True
+    )
+    county2ba = county2ba.drop(columns=["index"])
+    county2ba.set_index(["county", "state"], inplace=True)
+
+    # US Counties Shapefile
+    counties_shp_file = gpd.read_file(get_full_filename("cb_2020_us_county_500k.zip"))
+    # Remove AK, HI, PR, etc.
+    counties_shp_file = counties_shp_file[["NAME", "STUSPS", "geometry"]].rename(
+        columns={"NAME": "county", "STUSPS": "state"}
+    )
+    lower48 = counties_shp_file.loc[
+        counties_shp_file["state"].isin(states_to_map_with_counties)
+    ]
+    lower48.set_index(["county", "state"], inplace=True)
+    counties_with_ba = lower48.join(county2ba)
+    # Since these all seem to be in the pandhandle, map them to ERCOT north
+    counties_with_ba.loc[counties_with_ba["BA"] == "ERCOT", "BA"] = "ERCO-N"
```

Comment:
Now that we removed the `"ERCOT"` entries from `BA_County_map.json`, does this line have any effect?

Reply:
The six counties in `"ERCOT"` used to be assigned to `"ERCO-N"` in the python codes.  They are now assigned to `"ERCO-N"` in the ba_County_map.json.   The final map is not impacted.